spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/core/AwsClientCustomizer.java

```diff
@@ -34,6 +36,11 @@ default ClientOverrideConfiguration overrideConfiguration() {
 		return null;
 	}
 
+	@Nullable
+	default ClientOverrideConfiguration.Builder overrideConfigurationBuilder() {
```

Comment:
@maciejwalkowiak I was thinking and checking if we can parameterize this somehow through properties. Tbh I am not seeing a clean way to do it, there are just too many options, and some of them you can't even describe in properties.
Maybe we can support just some simple ones as 
```
    private final Duration apiCallAttemptTimeout;
    private final Duration apiCallTimeout;
```
and apply them in the builder if not null. If not we don't apply to the builder anything besides our `applyUserAgent`.